nixos/modules/services/web-apps/lasuite-docs.nix

```
@@ -0,0 +1,576 @@
+    PrivateUsers = true;
+    ProcSubset = "pid";
+    ProtectClock = true;
+    ProtectControlGroups = true;
+    ProtectHome = true;
+    ProtectHostname = true;
+    ProtectKernelLogs = true;
+    ProtectKernelModules = true;
+    ProtectKernelTunables = true;
+    ProtectProc = "invisible";
+    ProtectSystem = "strict";
+    RemoveIPC = true;
+    RestrictAddressFamilies = [
+      "AF_INET"
+      "AF_INET6"
+      "AF_UNIX"
+    ];
+    RestrictNamespaces = true;
+    RestrictRealtime = true;
+    RestrictSUIDSGID = true;
+    SystemCallArchitectures = "native";
+    UMask = "0077";
+  };
+
+  preloadSecrets = ''
+    ${optionalString (
+      cfg.s3.accessKeyIDPath != null
+    ) "export AWS_S3_ACCESS_KEY_ID=$(cat ${cfg.s3.accessKeyIDPath})"}
+    ${optionalString (
+      cfg.s3.secretAccessKeyPath != null
+    ) "export AWS_S3_SECRET_ACCESS_KEY=$(cat ${cfg.s3.secretAccessKeyPath})"}
+    ${optionalString (
+      cfg.oidc.clientSecretPath != null
+    ) "export OIDC_RP_CLIENT_SECRET=$(cat ${cfg.oidc.clientSecretPath})"}
+    ${optionalString (
+      cfg.collaborationServer.serverSecretPath != null
+    ) "export COLLABORATION_SERVER_SECRET=$(cat ${cfg.collaborationServer.serverSecretPath})"}
+    ${optionalString (
+      cfg.collaborationServer.yproviderApiKeyPath != null
+    ) "export Y_PROVIDER_API_KEY=$(cat ${cfg.collaborationServer.yproviderApiKeyPath})"}
+  '';
+in
+{
+  options.services.lasuite-docs = {
+    enable = mkEnableOption "SuiteNumérique Docs";
+
+    backendPackage = mkPackageOption pkgs "lasuite-docs" { };
+
+    frontendPackage = mkPackageOption pkgs "lasuite-docs-frontend" { };
+
+    bind = mkOption {
+      type = types.str;
+      default = "127.0.0.1:8000";
+      description = ''
+        The server socket used to bind.
+      '';
+    };
+
+    enableNginx = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Enable Nginx as a proxy server.
+      '';
+    };
+
+    secretKeyPath = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = ''
+        Path to the Django secret key.
+      '';
+    };
+
+    s3 = {
+      url = mkOption {
+        type = types.str;
+        description = ''
+          URL of the s3 bucket.
+        '';
+      };
+
+      accessKeyIDPath = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Path to the access key ID of the bucket.
+        '';
+      };
+
+      secretAccessKeyPath = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Path to the secret access key of the bucket.
+        '';
+      };
+    };
+
+    oidc = {
+      clientSecretPath = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Path to the client secret of the client of OIDC.
+        '';
+      };
+    };
+
+    database = {
```

```suggestion
    postgresql = {
```
Would be more consistent, no?